src/System/Metrics/Protocol/Forwarder.hs

```diff
@@ -35,17 +35,19 @@ ekgForwarderPeer
   :: Monad m
   => EKGForwarder req resp m a
   -> Peer (EKGForward req resp) 'AsServer 'StIdle m a
-ekgForwarderPeer EKGForwarder{..} =
-  -- In the 'StIdle' state the forwarder is awaiting a request message
-  -- from the acceptor.
-  Await (ClientAgency TokIdle) $ \case
-    -- The acceptor sent us a request for new metrics, so now we're
-    -- in the 'StBusy' state which means it's the forwarder's turn to send
-    -- a reply.
-    MsgReq req -> Effect $ do
-      (resp, next) <- recvMsgReq req
-      return $ Yield (ServerAgency TokBusy) (MsgResp resp) (ekgForwarderPeer next)
+ekgForwarderPeer EKGForwarder{..} = go
+  where
+    go =
+      -- In the 'StIdle' state the forwarder is awaiting a request message
+      -- from the acceptor.
+      Await (ClientAgency TokIdle) $ \case
+        -- The acceptor sent us a request for new metrics, so now we're
+        -- in the 'StBusy' state which means it's the forwarder's turn to send
+        -- a reply.
+        MsgReq req -> Effect $ do
+          resp <- recvMsgReq req
+          return $ Yield (ServerAgency TokBusy) (MsgResp resp) go
```

Comment:
This doesn't really change when `resp` is evaluated to WHNF, maybe you'd like to add a `!`?

Reply:
Does that really change anything?
1. `resp` hasn't shown to be leaky.
2. `resp` is basically a `[(MetricName, MetricValue)]`, so eval'ing to WHNF wouldn't make much difference
3. the list gets always fully evaluated when the response is sent over the wire

However, if you think it makes a difference, I can add the bang - you decide.

Reply:
That's fine then.  Thanks @mgmeier.